core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_Team.hpp

```diff
@@ -151,8 +151,18 @@ KOKKOS_INLINE_FUNCTION void parallel_reduce(
 #define KOKKOS_IMPL_ACC_REDUCE_TEAM_PRAGMA \
   vector vector_length(team_size* vector_length)
 #define KOKKOS_IMPL_ACC_REDUCE_TEAM_ITRS league_size* team_size* vector_length
-#define KOKKOS_IMPL_ACC_REDUCE_TEAM_LEAGUE_ID_INIT \
-  i / (team_size * vector_length)
+#ifdef KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE
+#define KOKKOS_IMPL_ACC_REDUCE_TEAM_MEMBER_INIT                          \
+  int league_rank = i / (team_size * vector_length);                     \
+  int team_rank   = i % (team_size * vector_length);                     \
```

Comment:
Why is it here that we need to compute the team rank ourselves?

Reply:
It is because when offloading to the host, the NVHPC OpenACC compiler parallelizes only the gang loop and ignores  worker/vector loops, which is not standard feature but the NVHPC-specific behavior.